modules/AOS_PDF_Templates/formLetter.php

```diff
@@ -80,45 +80,50 @@ static function LVPopupHtml($module)
-					<td scope="row" align="left"><b><a href="#" onclick="document.getElementById(\'popupDiv_ara\').style.display=\'none\';sListView.send_form(true, \'' . $_REQUEST['module'] .
-                    '\', \'index.php?templateID=' . $templateid . '&entryPoint=formLetter\',\'' . $app_strings['LBL_LISTVIEW_NO_SELECTED'] . '\');document.getElementById(\'popupDivBack_ara\').style.display=\'none\';">' . $template . '</a></b></td></tr>';
-            }
-            echo '<tr style="height:10px;"><tr><tr><td colspan="2"><button style=" display: block;margin-left: auto;margin-right: auto" onclick="document.getElementById(\'popupDivBack_ara\').style.display=\'none\';document.getElementById(\'popupDiv_ara\').style.display=\'none\';return false;">Cancel</button></td></tr>
-			</table>
-				</div>
-				<div id="popupDivBack_ara" onclick="this.style.display=\'none\';document.getElementById(\'popupDiv_ara\').style.display=\'none\';" style="top:0px;left:0px;position:fixed;height:100%;width:100%;background:#000000;opacity:0.5;display:none;vertical-align:middle;text-align:center;z-index:9998;">
-				</div>
-				<script>
-					function showPopup(){
-					if(sugarListView.get_checks_count() < 1)
-					{
-						alert(\'' . $app_strings['LBL_LISTVIEW_NO_SELECTED'] . '\');
-					}
-					else
-					{
-						var ppd=document.getElementById(\'popupDivBack_ara\');
-						var ppd2=document.getElementById(\'popupDiv_ara\');
-						if(ppd!=null && ppd2!=null){
-							ppd.style.display=\'block\';
-							ppd2.style.display=\'block\';
-						}else{
-							alert(\'Error!\');
-						}
-					}
-					}
-				</script>';
+            echo '    
+            
+            <div id="popupDiv_ara" class="modal fade" style="display: none;">
+               <div class="modal-dialog">
+                  <div class="modal-content">
+                     <div class="modal-header">
+                        <button type="button" class="close" data-dismiss="modal" aria-hidden="true">×</button>
+                        <h4 class="modal-title">'.$app_strings['LBL_SELECT_TEMPLATE'].'</h4>
+                     </div>
+                     <div class="modal-body">
+                        <div style="padding: 5px 5px; overflow: auto; height: auto;">
+                              <table width="100%" class="list view table default" cellspacing="0" cellpadding="0" border="0">
+                                 <tbody>';
+                                 foreach ($templates as $templateid => $template) {
+                                    echo '<tr height="20">
+                                            <td width="17" valign="center"><a href="#" onclick="$(\'#popupDiv_ara\').modal(\'hide\');sListView.send_form(true, \'' . $_REQUEST['module'] .
```

Comment:
Hey @urdhvatech Can you check out why Codacy is failing. What I can see its because you are using $_REQUEST['module'] & $_REQUEST['record'].  Processing form data without nonce verification.
https://app.codacy.com/app/salesagility/SuiteCRM/pullRequest?prid=2945728